lldb/source/Plugins/Process/scripted/ScriptedThread.cpp

```diff
@@ -229,6 +229,17 @@ bool ScriptedThread::CalculateStopInfo() {
         LLVM_PRETTY_FUNCTION, "Failed to get scripted thread stop info.", error,
         LLDBLog::Thread);
 
+  // If we're at a BreakpointSite, mark that we stopped there and
```

Comment:
Is this the code that's doing what the:

} else if (!signo) {
}

branch at 1883 in the old code was doing?

Reply:
This comment was meant to be on ProcessGDBRemote.cpp where I removed a section of code for a thread that had no stop signal/reason, but is sitting at a BreakpointSite.  The old comment on this:
```
        // If a thread is stopped at a breakpoint site, set that as the stop
        // reason even if it hasn't executed the breakpoint instruction yet.
        // We will silently step over the breakpoint when we resume execution
        // and miss the fact that this thread hit the breakpoint.
```

That's the old stepping behavior.  The new stepping behavior is if a thread stops at a BreakpointSite but has no signal/breakpoint hit reason, we call `Thread::SetThreadStoppedAtUnexecutedBP(pc)`.  This is done at the top of `ProcessGDBRemote::SetThreadStopInfo` already if the thread stops at a BreakpointSite but hasn't hit it, so there's no need for this block to still be around.